_episodes/04-building-containers.md

```diff
@@ -5,61 +5,80 @@ exercises: 30
 </a>
 
-# Build a container in a interactive session
-
-For building a container from a existing image in an interactive way, `build` provides a flag `--sandbox` that will
-create a writable directory.
-
-~~~bash
-singularity build --sandbox myCentOS7 docker://centos:centos7
+## Build a container in an interactive session
+
+While images contained in the `.sif` files are immutable objects, ideal for reproducibility, for building images in an
+interactive session is convenient the usage of a _sandbox_.
+The command `build` provides a flag `--sandbox` that will create a writable directory:
+```bash
+singularity build --sandbox myPython docker://python:3.9
+```
+The container name is `myPython`, and it has been initialized from the [Docker image](https://hub.docker.com/_/python)
+of Python using the tag 3.9.
+
+To initialize an interactive session use `shell`. And to write files within the container use the flag `--writable`:
+```bash
+singularity shell --writable myUbuntu
+```
+Once inside the container, you can make changes as needed for your work. For example, making
+[Uproot](https://uproot.readthedocs.io/en/latest/index.html) available
+```bash
+Singularity> pip install uproot awkward
+```
+Exit the container. Then confirm that uproot is available even after closing the shell:
+```bash
+singularity exec myPython python -c "import uproot; print(uproot.__doc__)"
+```
 ~~~
-{: .source}
+Uproot: ROOT I/O in pure Python and NumPy.
+...
+~~~
+{: .output}
 
-The sandbox name is `myCentOS7`, and it has been initialized from the [official Docker image](https://hub.docker.com/_/centos)
-of CentOS7.
-To initialize an interactive session use `shell`. And to write files within the sandbox directory  `--writable`.
-Additionally, the installation of new components will require superuser access:
+## Building a container with superuser permissions
 
-~~~bash
-sudo singularity shell --writable myCentOS7
+Some operations, like the installation of new components will require superuser access. To be superuser inside the container,
+you must be also outside (in the host machine). You can either open a shell as root, or use `sudo`:
 
+```bash
+$ sudo singularity shell --writable myCentOS7
+```
+The container name is `myCentOS7`, and it has been initialized from the [official Docker image](https://hub.docker.com/_/centos)
+of CentOS7. You can verify that opening a shell with `sudo` makes you root inside the container
+```bash
 Singularity> whoami
-~~~
-{: .source}
+```
 ~~~
 root
 ~~~
 {: .output}
 
-As an example, let's create a container with Pythia8 available using the `myCentOS7` sandbox.
+As a second example, let's create a container with Pythia8 available using the `myCentOS7` sandbox.
 First, we need to install the development tools (remember that in this interactive session we are superuser):
-
-~~~bash
+```bash
 Singularity> yum groupinstall 'Development Tools'
```

Comment:
btw, is `yum` explained somewhere? Not everyone might be familiar with that

Reply:
Good point! I will add a bit of information.